borsh/src/ser/mod.rs

```diff
@@ -201,6 +201,38 @@ impl BorshSerialize for String {
     }
 }
 
+#[cfg(feature = "bigdecimal")]
+impl BorshSerialize for bigdecimal::BigDecimal {
+    #[inline]
+    fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
+        let (bigint, exponent) = self.as_bigint_and_exponent();
```

Comment:
As I said above, we should normalize before serializing it:

```suggestion
        let (bigint, exponent) = self.normalized().as_bigint_and_exponent();
```

Reply:
Sorry, missed that